### PR TITLE
[10.0][MIG] Porting account_cost_center to V10

### PR DESCRIPTION
**Pull request by astirpe**

### CODE REVIEW

**Comment by astirpe** [comment]
@levkar @pmartineznetquest @yung-wang you may want to review this PR

**Comment by astirpe** [comment]
Needs fixing

**Comment by astirpe** [comment]
@thomaspaulb @hbrunn you may also be interested in this PR

**Review by AaronHForgeFlow — APPROVED** [review]
:+1: Functional test good so far.

File: account_cost_center/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016-2018 Onestein (<http://www.onestein.eu>)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+{
+    'name': 'Costcenter',
+    'summary': 'Cost center information for invoice lines',
+    'author': 'Onestein,Odoo Community Association (OCA)',
+    'license': 'AGPL-3',
+    'website': 'http://www.onestein.eu',
```
Comment (tbaden) [inline account_cost_center/__manifest__.py]: please change the website to https://github.com/OCA/account-financial-tools

File: account_cost_center/views/account_cost_center.xml [hunk 2]
```diff
@@ -0,0 +1,78 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+
+    <!-- Cost Center views -->
+    <record id="account_cost_center_form" model="ir.ui.view">
```
Comment (tbaden) [inline account_cost_center/views/account_cost_center.xml]: not approve relevant, but would be nice to have id pattern like

> For a view: <model_name>view<view_type>, where view_type is kanban, form, tree, search, ...

File: account_cost_center/views/account_cost_center.xml [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+                        </group>
+                    </group>
+                    <notebook name="notebook">
+                    </notebook>
+                </sheet>
+            </form>
+        </field>
+    </record>
+
+    <record id="account_cost_center_tree" model="ir.ui.view">
```
Comment (tbaden) [inline account_cost_center/views/account_cost_center.xml]: not approve relevant, but would be nice to have id pattern like

> For a view: <model_name>view<view_type>, where view_type is kanban, form, tree, search, ...

File: account_cost_center/views/account_cost_center.xml [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+        <field name="arch" type="xml">
+            <tree>
+                <field name="code"/>
+                <field name="name"/>
+                <field name="company_id" groups="base.group_multi_company"/>
+            </tree>
+        </field>
+    </record>
+
+    <record id="account_cost_center_filter" model="ir.ui.view">
```
Comment (tbaden) [inline account_cost_center/views/account_cost_center.xml]: not approve relevant, but would be nice to have id pattern like

> For a view: <model_name>view<view_type>, where view_type is kanban, form, tree, search, ...

File: account_cost_center/views/account_invoice.xml [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+            <xpath expr="//field[@name='invoice_line_ids']/tree/field[@name='account_id']" position="after">
+                <field name="cost_center_id"/>
+            </xpath>
+            <field name="invoice_line_ids" position="attributes">
+                <attribute name="context" operation="python_dict" key="cost_center_id">cost_center_id</attribute>
+            </field>
+        </field>
+    </record>
+
+    <record id="invoice_supplier_form_inherit" model="ir.ui.view">
```
Comment (tbaden) [inline account_cost_center/views/account_invoice.xml]: not approve relevant, but would be nice to have original id

> prefer naming the inherited view after the original over using a name which follows the guidelines.

```
<record id="original_id" model="ir.ui.view">
    <field name="inherit_id" ref="original_module.original_id"/>
```

File: account_cost_center/views/account_invoice_report.xml [hunk 6]
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+
+    <record id="view_account_invoice_report_search" model="ir.ui.view">
```
Comment (tbaden) [inline account_cost_center/views/account_invoice_report.xml:4]: not approve relevant, but would be nice to have id pattern like

> For a view: <model_name>view<view_type>, where view_type is kanban, form, tree, search, ...

File: account_cost_center/views/account_move.xml [hunk 7]
```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+
+    <record id="move_form_cost_center" model="ir.ui.view">
```
Comment (tbaden) [inline account_cost_center/views/account_move.xml]: not approve relevant, but would be nice to have original id

> prefer naming the inherited view after the original over using a name which follows the guidelines.

```
<record id="original_id" model="ir.ui.view">
    <field name="inherit_id" ref="original_module.original_id"/>
```

File: account_cost_center/views/account_move_line.xml [hunk 8]
```diff
@@ -0,0 +1,25 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+
+    <record id="move_line_form_cost_center" model="ir.ui.view">
```
Comment (tbaden) [inline account_cost_center/views/account_move_line.xml]: not approve relevant, but would be nice to have original id

> prefer naming the inherited view after the original over using a name which follows the guidelines.

```
<record id="original_id" model="ir.ui.view">
    <field name="inherit_id" ref="original_module.original_id"/>
```

File: account_cost_center/views/account_move_line.xml [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+        <field name="model">account.move.line</field>
+        <field name="inherit_id" ref="account.view_move_line_form"/>
+        <field name="arch" type="xml">
+            <field name="analytic_account_id" position="after">
+                <field name="cost_center_id"/>
+            </field>
+        </field>
+    </record>
+
+    <record id="move_line_tree_cost_center" model="ir.ui.view">
```
Comment (tbaden) [inline account_cost_center/views/account_move_line.xml]: not approve relevant, but would be nice to have original id

> prefer naming the inherited view after the original over using a name which follows the guidelines.

```
<record id="original_id" model="ir.ui.view">
    <field name="inherit_id" ref="original_module.original_id"/>
```

**Review by tbaden — COMMENTED** [review]
thanks for your work 👍 
i did some tests on local machine. everything works.

and a code review:
just the manifest part the relevant, the other ones are possible aesthetic changes

File: account_cost_center/views/account_invoice_report.xml (hunk 6 above)
Comment (astirpe) [reply account_cost_center/views/account_invoice_report.xml:4]: This one seems correct as it is :grin:

**Review by tbaden — APPROVED** [review]
code review: lgtm and thanks for the aesthetic changes 👍

**Review by AaronHForgeFlow — APPROVED** [review]

**Pull request merged**